src/compiler/checker.ts

```diff
@@ -6052,6 +6066,28 @@ namespace ts {
                 }
                 return Ternary.True;
             }
+
+            function constructorRelatedTo(sourceSignature: Signature, targetSignature: Signature, reportErrors: boolean) {
+                if (sourceSignature && targetSignature && sourceSignature.declaration && targetSignature.declaration) {

```

Comment:
There are cases where the declarations are `undefined`. I checked and ran the tests, which caused lots of errors.
But I've removed `sourceSignature` and `targetSignature`.